### PR TITLE
Feature/69 programme overview

### PR DESCRIPTION
**Pull request by kevinhowbrook**
This PR will put in place the fields for the programme page 'overview' section.

###  FE: The template changes to relatedcontent > relatedprogrammes
I've done this so we can keen related content for a more generic reason down the line, and the variable names and classes look specific to programmes. [link](https://github.com/torchbox/rca-wagtail-2019/pull/54/files#diff-cddd0aad862aa16e78fb3d87af5f10a6)

### BE: The context update for relatedprogrammes
I've done this [in this way](https://github.com/torchbox/rca-wagtail-2019/pull/54/files#diff-7e356f728d6d9ada8c4d7be20f54d5fcR358) to match the structure defined by FE in the template

### Concerns
My main concern at the moment is performance optimisation, I'd like to be sure we are making this page as fast as possible given it's going to have at least 30 more fields over new sections.

### CODE REVIEW

File: rca/programmes/models.py [hunk 1]
```diff
@@ -139,6 +154,76 @@ class ProgrammePage(BasePage):
...
 
+    # Programme Overview
+    programme_description_title = models.CharField(max_length=50, blank=True, null=True)
+    programme_description_subtitle = models.CharField(
+        max_length=100, blank=True, null=True
+    )
+    programme_description_copy = RichTextField(blank=True, null=True)
+    # Gallery
+    programme_gallery_title = models.CharField(max_length=125, blank=True, null=True)
+    programme_gallery = StreamField(
```
Comment (kevinhowbrook) [inline rca/programmes/models.py]: Make this a class based block in utils/or in blocks.py

File: rca/project_styleguide/templates/patterns/pages/programmes/programme_detail.html [hunk 2]
```diff
@@ -75,16 +82,15 @@ <h3 class="section__heading section__heading--secondary heading heading--four">A
             </section>
 
 
-
-            <section class="section bg bg--light">
+            <div class="section bg bg--light">
```
Comment (nicklee) [inline rca/project_styleguide/templates/patterns/pages/programmes/programme_detail.html]: This should be a section

Comment (nicklee) [reply rca/project_styleguide/templates/patterns/pages/programmes/programme_detail.html]: I've fixed this

File: rca/programmes/models.py [hunk 3]
```diff
@@ -155,11 +148,9 @@ class ProgrammePage(BasePage):
...
-    programme_description_title = models.CharField(max_length=50, blank=True, null=True)
-    programme_description_subtitle = models.CharField(
-        max_length=100, blank=True, null=True
-    )
-    programme_description_copy = RichTextField(blank=True, null=True)
+    programme_description_title = models.CharField(max_length=50, blank=True)
+    programme_description_subtitle = models.CharField(max_length=100, blank=True)
+    programme_description_copy = RichTextField(blank=True)
     # Gallery
     programme_gallery_title = models.CharField(max_length=125, blank=True, null=True)
```
Comment (richbrennan) [inline rca/programmes/models.py]: Remove `null=True`?

File: rca/programmes/models.py [hunk 4]
```diff
@@ -178,13 +169,12 @@ class ProgrammePage(BasePage):
...
         ],
         blank=True,
-        null=True,
     )
 
-    facilities_intro = models.CharField(blank=True, null=True, max_length=150)
-    facilities_copy = models.TextField(blank=True, null=True, max_length=400)
+    facilities_intro = models.CharField(blank=True, max_length=150)
+    facilities_copy = models.TextField(blank=True, max_length=400)
     facilities_link = models.URLField(blank=True, null=True)
```
Comment (richbrennan) [inline rca/programmes/models.py]: `URLField` is based on a `CharField` so don't need `null=True` https://docs.djangoproject.com/en/2.2/ref/models/fields/#urlfield

File: rca/programmes/models.py [hunk 5]
```diff
@@ -139,6 +147,71 @@ class ProgrammePage(BasePage):
...
+                "Link_to_person",
+                StructBlock([("name", CharBlock()), ("link", URLBlock())], icon="link"),
+            )
+        ],
+        blank=True,
+    )
+
+    contact_title = models.CharField(max_length=125, default="Contact us")
+    contact_text = models.TextField(blank=True)
+    contact_link_url = models.URLField(blank=True, null=True)
```
Comment (richbrennan) [inline rca/programmes/models.py]: `null=True` can be removed

File: rca/project_styleguide/templates/patterns/molecules/relatedcontent/relatedprogrammes.html [hunk 6]
```diff
@@ -0,0 +1,50 @@
+{% load wagtailcore_tags wagtailimages_tags %}
+
+<div class="related-content">
+    <div class="section__row grid">
+        <h2 class="related-content__heading heading heading--two">{{title}}</h2>
+    </div>
+
+    {% if related_sections %}
```
Comment (richbrennan) [inline rca/project_styleguide/templates/patterns/molecules/relatedcontent/relatedprogrammes.html]: Is `related_sections` correct, `related_programmes ` is being assigned in the context? https://github.com/torchbox/rca-wagtail-2019/pull/54/files#diff-7e356f728d6d9ada8c4d7be20f54d5fcR346

Comment (richbrennan) [reply rca/project_styleguide/templates/patterns/molecules/relatedcontent/relatedprogrammes.html]: Also, you don't need the `if` statement

File: rca/project_styleguide/templates/patterns/molecules/relatedcontent/relatedprogrammes.html [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+                                    {{related_item.title}}
+                                </a>
+                            </p>
+                            <p class="related-content__degree body body--two">{{related_item.degree_level}}</p>
+                        </div>
+                        <p class="related-content__copy body body--two">
+                            {{related_item.programme_description_subtitle}}
+                        </p>
+                    {% endfor %}
+                    {% for related_item in related_section.related_items %}
```
Comment (richbrennan) [inline rca/project_styleguide/templates/patterns/molecules/relatedcontent/relatedprogrammes.html]: Do you need to repeat the loop here?

**Pull request merged**